main/views.py

```diff
@@ -449,6 +452,27 @@ def delete(self, request, *args, **kwargs):
         return super(DeleteView, self).delete(request, *args, **kwargs)
 
 
+class UserMediaView(LoginRequiredMixin, generic.View):
+    """Respond with a given user uploaded file if the request
+    is logged in. Makes it so that user uploads are no longer
+    public."""
+
+    def get_response_file(self, filename):
+        filepath = os.path.join(settings.MEDIA_ROOT, filename)
+        try:
```

Comment:
```suggestion
if filename in os.listdir(settings.MEDIA_ROOT):  # check if file exists in media root
```

Would this be sufficient sanitization?